component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -129,8 +129,18 @@ public boolean canHandle(HttpServletRequest request) {
         return ((StringUtils.isNotEmpty(request.getParameter(EmailOTPAuthenticatorConstants.RESEND))
                 && StringUtils.isEmpty(request.getParameter(EmailOTPAuthenticatorConstants.CODE)))
                 || StringUtils.isNotEmpty(request.getParameter(EmailOTPAuthenticatorConstants.CODE))
-                || StringUtils.isNotEmpty(request.getParameter(EmailOTPAuthenticatorConstants.EMAIL_ADDRESS))
-                || StringUtils.isNotEmpty(request.getParameter(EmailOTPAuthenticatorConstants.USER_NAME)));
+                || StringUtils.isNotEmpty(request.getParameter(EmailOTPAuthenticatorConstants.EMAIL_ADDRESS)));
+    }
+
+    @Override
+    public boolean canHandleMultiOptionRequest(HttpServletRequest request, AuthenticationContext context) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Inside EmailOTPAuthenticator canHandleMultiOptionRequest method");
+        }
+        return canHandle(request) ||
+                (StringUtils.isNotBlank(request.getParameter(EmailOTPAuthenticatorConstants.USER_NAME)) &&
+                        getName().equalsIgnoreCase(context.getCurrentAuthenticator()));
```

Comment:
This check is there but for basic and idf authenticators  currentAuthenticator is null